### PR TITLE
Add logging to user management endpoints

### PR DESCRIPTION
**Pull request by ivov**
WIP

### CODE REVIEW

File: packages/cli/src/UserManagement/routes/users.ts [hunk 1]
```diff
@@ -86,38 +94,52 @@ export function usersNamespace(this: N8nApp): void {
...
+						resp.error = `Email could not be sent`;
+					}
+					return resp;
+				}),
+			);
+
+			Logger.debug(
+				usersPendingSetup.length > 1
+					? `Sent ${usersPendingSetup.length} emails successfully`
+					: `Sent 1 email successfully`,
```
Comment (BHesseldieck) [inline packages/cli/src/UserManagement/routes/users.ts:141]: let's be more specific and say `invite emails`

File: packages/cli/src/UserManagement/routes/users.ts [hunk 2]
```diff
@@ -129,26 +151,22 @@ export function usersNamespace(this: N8nApp): void {
...
 				throw new ResponseHelper.ResponseError('Invalid invite URL', undefined, 400);
 			}
 
 			const inviter = users.find((user) => user.id === inviterId);
 
 			if (!inviter || !inviter.email || !inviter.firstName) {
-				LoggerProxy.error('Invalid invite URL - inviter does not have email set', {
-					inviterId,
-					inviteeId,
+				Logger.error('Missing inviter or inviter email or inviter firstName', {
```
Comment (BHesseldieck) [inline packages/cli/src/UserManagement/routes/users.ts:168]: `Inviter doesn't exist or is not setup`

File: packages/cli/src/UserManagement/routes/users.ts [hunk 3]
```diff
@@ -178,20 +196,23 @@ export function usersNamespace(this: N8nApp): void {
 			};
 
 			if (!inviterId || !inviteeId || !firstName || !lastName || !password) {
-				throw new ResponseHelper.ResponseError('Invalid payload', undefined, 400);
+				Logger.debug('Missing property or properties in payload', { payload: req.body });
```
Comment (BHesseldieck) [inline packages/cli/src/UserManagement/routes/users.ts:199]: `Missing properties in payload`

File: packages/cli/src/UserManagement/routes/index.ts [hunk 4]
```diff
@@ -77,10 +82,10 @@ export function addRoutes(this: N8nApp, ignoredEndpoints: string[], restEndpoint
 		return passport.authenticate('jwt', { session: false })(req, res, next);
 	});
 
-	this.app.use((req: Request, res: Response, next: NextFunction) => {
+	this.app.use((req: Request | AuthenticatedRequest, res: Response, next: NextFunction) => {
 		// req.user is empty for public routes, so just proceed
 		// owner can do anything, so proceed as well
-		if (req.user === undefined || (req.user && (req.user as User).globalRole.name === 'owner')) {
+		if (!req.user || (isAuthenticatedRequest(req) && req.user.globalRole.name === 'owner')) {
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/cli/src/UserManagement/routes/index.ts:88]: On the contrary, they are opposite cases. The guard checks that `user` does exist.

File: packages/cli/src/UserManagement/routes/passwordReset.ts [hunk 5]
```diff
@@ -88,6 +104,10 @@ export function passwordResetNamespace(this: N8nApp): void {
 			const { token: resetPasswordToken, userId, password } = req.body;
 
 			if (!resetPasswordToken || !userId || !password) {
+				Logger.debug('User ID or password or reset password token missing from payload', {
+					userId: req.user.id,
```
[the remark this answers is not in the document]
Comment (ivov) [reply packages/cli/src/UserManagement/routes/passwordReset.ts:108]: Had been fixed already in newer branch

File: packages/cli/src/UserManagement/routes/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (ivov) [reply packages/cli/src/UserManagement/routes/index.ts:88]: Not exactly. The type guard specifies not only that `req.user` is not `undefined` but also that `req.user` is of type `User`, which allows for role access without the assertion. Without the type guard, in the second half we know that `req.user` is not `undefined` but not what type it is.

**Pull request closed without merging**